lib/Doctrine/DBAL/Exception/LockWaitTimeoutException.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Exception;
+
+/**
+ * Exception for a lock wait timeout error of a transaction detected in the driver.
+ *
+ * @author Tobias Schultze <http://tobion.de>
+ * @link   www.doctrine-project.org
+ * @since  2.6
+ */
+class LockWaitTimeoutException extends ServerException implements RetryableException
```

Comment:
@Tobion is there a reason why we distinguish `DeadLockException` and `LockWaitTimeoutException`? Are semantics different so that we need two distinct exception classes for this? I have seen the term `lock wait timeout` only in MySQL context so far and it tells more or less the same thing as `deadlock` in other DBMS, no? If there is no real benefit in having two exception classes here I would propose dropping this one. What do you think?

/cc @zeroedin-bill @Ocramius